src/main/java/edu/hm/hafner/analysis/parser/DScannerParser.java

```diff
@@ -0,0 +1,100 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.ParsingException;
+import edu.hm.hafner.analysis.ReaderFactory;
+import edu.hm.hafner.analysis.Report;
+import edu.hm.hafner.analysis.Severity;
+
+/**
+ * Parser report in JSON format as exported by DScanner.
+ *
+ * @author Andre Pany
+ */
+public class DScannerParser extends JsonBaseParser {
+    private static final long serialVersionUID = -3396574381502866972L;
+    private static final String ISSUES = "issues";
+    private static final String KEY = "key";
+    private static final String LINE = "line";
+    private static final String COLUMN = "column";
+
+    @Override
+    public boolean accepts(final ReaderFactory readerFactory) {
+        return readerFactory.getFileName().endsWith(".json");
+    }
+
+    @Override
+    public Report parse(final ReaderFactory readerFactory) throws ParsingException {
+        try (Reader reader = readerFactory.create()) {
+            JSONObject jsonReport = (JSONObject)new JSONTokener(reader).nextValue();
+
+            Report report = new Report();
+            if (jsonReport.has(ISSUES)) {
+                JSONArray issues = jsonReport.getJSONArray(ISSUES);
+                StreamSupport.stream(issues.spliterator(), false)
+                        .filter(o -> o instanceof JSONObject)
+                        .map(o -> convertToIssue((JSONObject)o))
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .forEach(report::add);
+            }
+            return report;
+        }
+        catch (IOException | JSONException e) {
+            throw new ParsingException(e);
+        }
+    }
+
+    @Override
+    protected Optional<Issue> convertToIssue(final JSONObject jsonIssue) {
+        IssueBuilder builder = new IssueBuilder();
+        if (jsonIssue.has(KEY)) {
+            String key = jsonIssue.getString(KEY);
+            builder.setCategory(key);
+            builder.setSeverity(getSeverityByKey(key));
+        }
+        if (jsonIssue.has(FILE_NAME)) {
+            builder.setFileName(jsonIssue.getString(FILE_NAME));
+        }
+        if (jsonIssue.has(LINE)) {
+            builder.setLineStart(jsonIssue.getInt(LINE));
+        }
+        if (jsonIssue.has(COLUMN)) {
+            builder.setColumnStart(jsonIssue.getInt(COLUMN));
+        }
+        if (jsonIssue.has(MESSAGE)) {
+            builder.setMessage(jsonIssue.getString(MESSAGE));
+        }
+        return builder.buildOptional();
+    }
+
+    private Severity getSeverityByKey(final String key) {
+        String[] parts = key.split("\\.");
```

Comment:
Would make more safe to use ` key.split("\\.", -1);`, see https://ci.jenkins.io/job/Plugins/job/analysis-model/view/change-requests/job/PR-189/5/error-prone/new/source.bd37f77a-752a-4696-a046-73324be6f7db/#83.